### PR TITLE
(#2556) Expand PowerShell aliases

### LINKED ISSUE #2556: ChocolateyTabExpansion.ps1 can cause slow auto-complete in PowerShell
### What You Are Seeing?
PowerShell autocomplete slows to a crawl when ChocolateyTabExpansion.ps1 is loaded and default "where" alias for Where-Object does not exist.

### What is Expected?
loading of "ChocolateyTabExpansion" should not slow Tab autocomplete (should not rely on alias).

Standard PowerShell programming practices is to not use aliases in code/modules, only on the command line.

### How Did You Get This To Happen? (Steps to Reproduce)

1. Added the following to my PSProfile:

   If (Test-Path Alias:diff ) {Remove-Item Alias:diff -Force}

2. Load PowerShell and try and hit tab to autocomplete

Normal behaviour restored by either commenting out loading of Choco module OR commenting out line to remove where alias.

### Output Log
No Output (problem is not with exe)

I fixed this manually myself by editing "ChocolateyTabExpansion.ps1" and changed all 12 occurences of "| where" to "| Where-Object" and problem was solved.

diff follows below

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
## Description Of Changes

This PR expands all of the PowerShell aliases in helper functions and
other PowerShell files included in the repository, excluding the pester
tests in the `tests` folder.

## Motivation and Context

Using full cmdlets is best scripting practice, as it prevents issues if an alias is not available or removed.

## Testing

Checking that #2556 is fixed was tested by:
- Adding this to my powershell profile: `If (Test-Path Alias:where ) {Remove-Item Alias:where -Force}`
- Opening a new powershell window. 
- Importing the modified `chocolateyProfile.psm1`
- Checking that tab completion works normally.

I used this command to ensure that no aliases where missed: `Get-Childitem -Path . -Exclude "tests" | Invoke-ScriptAnalyzer -IncludeRule 'PSAvoidUsingCmdletAliases' -Recurse`

## Change Types Made

* [x] Bug fix (non-breaking change)
* [ ] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)

## Related Issue

Fixes #2556

## Change Checklist

* [ ] Requires a change to the documentation
* [ ] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [x] All new and existing tests passed.
* [x] PowerShell v2 compatibility checked.

### CODE REVIEW

**Comment by corbob** [comment]
One thing for us to be aware of (for the future) is that (some) PSSA rules don't like Pester tests. We'll either want to exclude them from the analyzer, or try to test them with a different set of rule.

Particularly when we get to merging #2598

**Comment by TheCakeIsNaOH** [comment]
@corbob I've added a settings file, so it is only checking aliases at the moment. Other rules can get enabled later, after checking they will not interfere with the Pester tests (and fixing/excluding anything that the rule finds).

This does not work fully until it is merged into the repository, as the `secrets.GITHUB_TOKEN` is not available until this workflow is merged in. I have a test PR here which has the comment function is working: https://github.com/TheCakeIsNaOH/choco/pull/3

**Comment by TheCakeIsNaOH** [comment]
After discussion with @gep13, this PR has been narrowed in scope to focus exclusively on expanding aliases. In the future, once Cake has been implemented for this repository, a PSScriptAnalyzer check can be added to the Cake recipe.

**Comment by gep13** [comment]
@vexx32 @AdmiringWorm @corbob @JPRuskin @Windos PowerShell folks, what are your thoughts on this PR?

Looking at it, the only one I am concerned about is the switch from `Get-ProcessorBits` to `Get-OSArchitectureWidth`.  This is _our_ CmdLet, which looks like it was renamed at some point.  What are our thoughts on removing the alias usage for this one?

**Comment by AdmiringWorm** [comment]
> @vexx32 @AdmiringWorm @corbob @JPRuskin @Windos PowerShell folks, what are your thoughts on this PR?
> 
> Looking at it, the only one I am concerned about is the switch from `Get-ProcessorBits` to `Get-OSArchitectureWidth`. This is _our_ CmdLet, which looks like it was renamed at some point. What are our thoughts on removing the alias usage for this one?

In this specific case, I believe it is fine.
If it had been a call in a helper outside of Chocolatey CLI we would need to keep it as is, but not in this case.

Since both the file the call is located in, and the helper script file itself will be provided as part of the same version of Chocolatey CLI there are no functional changes to it. Not even backward breaking changes either.

File: nuget/chocolatey/tools/chocolateysetup.psm1 [hunk 1]
```diff
@@ -504,9 +504,9 @@ param(
     }
 
     if (!(Test-Path("$chocoPkgDirectory\chocolatey.nupkg"))) {
-      $chocoPkg = Get-ChildItem "$thisScriptFolder/../../" | ?{$_.name -match "^chocolatey.*nupkg" } | Sort name -Descending | Select -First 1
+      $chocoPkg = Get-ChildItem "$thisScriptFolder/../../" | Where-Object{$_.name -match "^chocolatey.*nupkg" } | Sort-Object name -Descending | Select-Object -First 1
```
Comment (vexx32) [inline nuget/chocolatey/tools/chocolateysetup.psm1:507]:
```suggestion
      $chocoPkg = Get-ChildItem "$thisScriptFolder/../../" |
          Where-Object { $_.name -match "^chocolatey.*nupkg" } |
          Sort-Object name -Descending |
          Select-Object -First 1
```

File: src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1 [hunk 2]
```diff
@@ -80,31 +80,31 @@ function script:chocoCommands($filter) {
...
+            Where-Object { $_ -like "$filter*" }
     }
 
     $cmdList #| sort
 }
 
 function script:chocoLocalPackages($filter) {
     if ($filter -ne $null -and $filter.StartsWith(".")) { return; } #file search
-    @(& $script:choco list $filter -lo -r --id-starts-with) | %{ $_.Split('|')[0] }
+    @(& $script:choco list $filter -lo -r --id-starts-with) | ForEach-Object{ $_.Split('|')[0] }
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1:93]:
```suggestion
    @(& $script:choco list $filter -lo -r --id-starts-with) | ForEach-Object { $_.Split('|')[0] }

```

File: src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1 [hunk 3]
```diff
@@ -80,31 +80,31 @@ function script:chocoCommands($filter) {
...
 }
 
 function script:chocoLocalPackagesUpgrade($filter) {
     if ($filter -ne $null -and $filter.StartsWith(".")) { return; } #file search
-    @('all|') + @(& $script:choco list $filter -lo -r --id-starts-with) | where { $_ -like "$filter*" } | %{ $_.Split('|')[0] }
+    @('all|') + @(& $script:choco list $filter -lo -r --id-starts-with) | Where-Object { $_ -like "$filter*" } | ForEach-Object{ $_.Split('|')[0] }
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1:98]:
```suggestion
    @('all|') + @(& $script:choco list $filter -lo -r --id-starts-with) |
        Where-Object { $_ -like "$filter*" } |
        ForEach-Object { $_.Split('|')[0] }
```

File: src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1 [hunk 4]
```diff
@@ -80,31 +80,31 @@ function script:chocoCommands($filter) {
...
 }
 
 function script:chocoRemotePackages($filter) {
     if ($filter -ne $null -and $filter.StartsWith(".")) { return; } #file search
-    @('packages.config|') + @(& $script:choco search $filter --page='0' --page-size='30' -r --id-starts-with --order-by-popularity) | where { $_ -like "$filter*" } | %{ $_.Split('|')[0] }
+    @('packages.config|') + @(& $script:choco search $filter --page='0' --page-size='30' -r --id-starts-with --order-by-popularity) | Where-Object { $_ -like "$filter*" } | ForEach-Object{ $_.Split('|')[0] }
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1:103]:
```suggestion
    @('packages.config|') + @(& $script:choco search $filter --page='0' --page-size='30' -r --id-starts-with --order-by-popularity) |
        Where-Object { $_ -like "$filter*" } |
        ForEach-Object { $_.Split('|')[0] }

```

File: src/chocolatey.resources/helpers/functions/Install-ChocolateyVsixPackage.ps1 [hunk 5]
```diff
@@ -157,11 +157,11 @@ param(
         if ([System.IntPtr]::Size -eq 4)
         {
             <# 32bits system case #>
-            $versions=(get-ChildItem HKLM:SOFTWARE\Microsoft\VisualStudio -ErrorAction SilentlyContinue | ? { ($_.PSChildName -match "^[0-9\.]+$") } | ? {$_.property -contains "InstallDir"} | sort {[int]($_.PSChildName)} -descending)
+            $versions=(get-ChildItem HKLM:SOFTWARE\Microsoft\VisualStudio -ErrorAction SilentlyContinue | Where-Object { ($_.PSChildName -match "^[0-9\.]+$") } | Where-Object {$_.property -contains "InstallDir"} | Sort-Object {[int]($_.PSChildName)} -descending)
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/functions/Install-ChocolateyVsixPackage.ps1:160]:
```suggestion
            $versions = Get-ChildItem HKLM:SOFTWARE\Microsoft\VisualStudio -ErrorAction SilentlyContinue |
                Where-Object { ($_.PSChildName -match "^[0-9\.]+$") } |
                Where-Object { $_.Property -contains "InstallDir" } |
                Sort-Object { [int]($_.PSChildName) } -Descending
```

File: src/chocolatey.resources/helpers/functions/Install-ChocolateyVsixPackage.ps1 [hunk 6]
```diff
@@ -157,11 +157,11 @@ param(
         if ([System.IntPtr]::Size -eq 4)
         {
             <# 32bits system case #>
-            $versions=(get-ChildItem HKLM:SOFTWARE\Microsoft\VisualStudio -ErrorAction SilentlyContinue | ? { ($_.PSChildName -match "^[0-9\.]+$") } | ? {$_.property -contains "InstallDir"} | sort {[int]($_.PSChildName)} -descending)
+            $versions=(get-ChildItem HKLM:SOFTWARE\Microsoft\VisualStudio -ErrorAction SilentlyContinue | Where-Object { ($_.PSChildName -match "^[0-9\.]+$") } | Where-Object {$_.property -contains "InstallDir"} | Sort-Object {[int]($_.PSChildName)} -descending)
         }
         else
         {
-            $versions=(get-ChildItem HKLM:SOFTWARE\Wow6432Node\Microsoft\VisualStudio -ErrorAction SilentlyContinue | ? { ($_.PSChildName -match "^[0-9\.]+$") } | ? {$_.property -contains "InstallDir"} | sort {[int]($_.PSChildName)} -descending)
+            $versions=(get-ChildItem HKLM:SOFTWARE\Wow6432Node\Microsoft\VisualStudio -ErrorAction SilentlyContinue | Where-Object { ($_.PSChildName -match "^[0-9\.]+$") } | Where-Object {$_.property -contains "InstallDir"} | Sort-Object {[int]($_.PSChildName)} -descending)
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/functions/Install-ChocolateyVsixPackage.ps1:164]:
```suggestion
            $versions = Get-ChildItem HKLM:SOFTWARE\Wow6432Node\Microsoft\VisualStudio -ErrorAction SilentlyContinue |
                Where-Object { ($_.PSChildName -match "^[0-9\.]+$") } |
                Where-Object { $_.Property -contains "InstallDir" } |
                Sort-Object { [int]($_.PSChildName) } -Descending
```

**Review by vexx32 — COMMENTED** [review]
I think these changes are fine. Though as long as we're doing something with this file it probably wouldn't hurt to clean up where we're editing just a little bit. This file is a bit of a mess 😔 (suggestions below)

**Comment by Windos** [comment]
Agree with @AdmiringWorm and @vexx32 (and very 👍👍 Joel's additional suggested changed). Chaning away from the `Get-ProcessorBits` alias makes sense (and makes following the trail to the actual function that much easier.)

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@TheCakeIsNaOH thanks for getting this updated!

**Pull request merged**